### DISCUSSION TITLE
About bookmark hash

### DISCUSSION THREAD

**Discussion by Alexplose** [elsa-workflows/elsa-core#1274] [Q&A]
Hi,
I'm having trouble using TriggerSignalAsync to resume an existing workflow.

Workflow is supended to a SignalReceived activity, and has a specific correlationId.

When using : 
await _signaler.TriggerSignalAsync("WaitingPayVariablesValidation", correlationId: instance.CorrelationId);
The workflow is not resumed.
But when using 
await _signaler.TriggerSignalAsync("WaitingPayVariablesValidation", instanceId: instance.Id);
The workflow is resumed.

In sql server profiler I've seen when using correlationId in TriggerSignalAsync that the bookmark hash doesn't match the targeted signal bookmark.
But when using the instanceId parameter the hash matches.

How do I trigger a workflow using correlationId ?

Also there is only one workflow that has this correlationId.

Help appreciated.

**Comment by sfmskywalker** [comment]
There was a bug with resuming workflow using correlation ID and signals that was recently fixed (one or two days ago).

Are you using the latest 2.1 packages? If so, then a repro would be much appreciated so I can investigate.

**Reply by Alexplose** [reply]
I was using 2.0.0.159 and now updating to 2.1.0.443 to see if it's working, I'll let you know.

Thanks

**Reply by sfmskywalker** [reply]
Cool. Just a heads up, there’s a bug with invoking HTTP Endpoints when used in combination with BasePath configured (e.g. /workflows) causing 404. A fix was pushed since (2.1.1), but I’m not sure it’s on NuGet yet.

**Reply by sfmskywalker** [reply]
The afore mentioned bug has been fixed.
